Argument of type 'typeof fastifyOpenTelemetry' is not assignable to parameter of type 'FastifyPluginCallback<{ wrapRoutes: boolean; }, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger> | FastifyPluginAsync<...> | Promise<...> | Promise<...>'.

reproducable here:
https://github.com/jmikedupont2/fastify-eliza-vercel/actions/runs/12624370786/job/35174543578#step:5:67

```
> tsc

Error: src/plugins/tracing.ts(13,22): error TS2769: No overload matches this call.
  Overload 1 of 6, '(plugin: FastifyPluginCallback<{ wrapRoutes: boolean; }, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger>, opts: FastifyRegisterOptions<...>): FastifyInstance<...> & ... 1 more ... & { ...; }', gave the following error.
    Argument of type 'typeof fastifyOpenTelemetry' is not assignable to parameter of type 'FastifyPluginCallback<{ wrapRoutes: boolean; }, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger>'.
      Type 'typeof fastifyOpenTelemetry' provides no match for the signature '(instance: FastifyInstance<RawServerDefault, IncomingMessage, ServerResponse<IncomingMessage>, FastifyBaseLogger, FastifyTypeProvider>, opts: { ...; }, done: (err?: Error | undefined) => void): void'.
  Overload 2 of 6, '(plugin: FastifyPluginAsync<{ wrapRoutes: boolean; }, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger>, opts: FastifyRegisterOptions<...>): FastifyInstance<...> & ... 1 more ... & { ...; }', gave the following error.
    Argument of type 'typeof fastifyOpenTelemetry' is not assignable to parameter of type 'FastifyPluginAsync<{ wrapRoutes: boolean; }, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger>'.
      Type 'typeof fastifyOpenTelemetry' provides no match for the signature '(instance: FastifyInstance<RawServerDefault, IncomingMessage, ServerResponse<IncomingMessage>, FastifyBaseLogger, FastifyTypeProvider>, opts: { ...; }): Promise<...>'.
  Overload 3 of 6, '(plugin: FastifyPluginCallback<{ wrapRoutes: boolean; }, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger> | FastifyPluginAsync<...> | Promise<...> | Promise<...>, opts: FastifyRegisterOptions<...>): FastifyInstance<...> & ... 1 more ... & { ...; }', gave the following error.
    Argument of type 'typeof fastifyOpenTelemetry' is not assignable to parameter of type 'FastifyPluginCallback<{ wrapRoutes: boolean; }, RawServerDefault, FastifyTypeProvider, FastifyBaseLogger> | FastifyPluginAsync<...> | Promise<...> | Promise<...>'.
 ELIFECYCLE  Command failed with exit code 2.
Error: The operation was canceled.
0s
Post job cleanup.
Pruning is unnecessary.
```

--- a/.eslintrc
+++ b/.eslintrc
@@ -4,12 +4,12 @@
   "overrides": [{
     "files": ["*.ts"],
     "extends": [
-      "eslint:recommended",
-      "plugin:@typescript-eslint/eslint-recommended",
-      "plugin:@typescript-eslint/recommended",
-      "standard"
+      "plugin:@typescript-eslint/recommended"
     ],
     "parser": "@typescript-eslint/parser",
-    "plugins": ["@typescript-eslint"]
+    "plugins": ["@typescript-eslint"],
+    "rules": {
+      "no-use-before-define": "off"
+    }
   }]
 }

--- a/fastify-opentelemetry.d.ts
+++ b/fastify-opentelemetry.d.ts
@@ -12,6 +12,8 @@ type ReqInstance = {
   extract: <Carrier> (carrier: Carrier, getter?: TextMapGetter) => Context,
 }
 
+type FastifyOpenTelemetry = FastifyPluginAsync<fastifyOpenTelemetry.OpenTelemetryPluginOptions>
+
 declare module 'fastify' {
   interface FastifyRequest {
     openTelemetry: () => ReqInstance
@@ -39,9 +41,11 @@ declare namespace fastifyOpenTelemetry {
     propagateToReply?: boolean,
   }
 
-  export const fastifyOpenTelemetry: FastifyPluginAsync<OpenTelemetryPluginOptions>
+  export const fastifyOpenTelemetry: FastifyOpenTelemetry
 
   export { fastifyOpenTelemetry as default }
 }
+
+declare function fastifyOpenTelemetry(...params: Parameters<FastifyOpenTelemetry>): ReturnType<FastifyOpenTelemetry>
 
 export = fastifyOpenTelemetry

--- a/package.json
+++ b/package.json
@@ -2,6 +2,7 @@
   "name": "@autotelic/fastify-opentelemetry",
   "version": "0.22.0",
   "description": "A Fastify plugin for OpenTelemetry",
+  "type": "commonjs",
   "main": "index.js",
   "files": [
     "fastify-opentelemetry.d.ts"

